core/src/test/java/io/questdb/test/griffin/AllTablePartitionsTest.java

```diff
@@ -0,0 +1,175 @@
+/*******************************************************************************
+ *     ___                  _   ____  ____
+ *    / _ \ _   _  ___  ___| |_|  _ \| __ )
+ *   | | | | | | |/ _ \/ __| __| | | |  _ \
+ *   | |_| | |_| |  __/\__ \ |_| |_| | |_) |
+ *    \__\_\\__,_|\___||___/\__|____/|____/
+ *
+ *  Copyright (c) 2014-2019 Appsicle
+ *  Copyright (c) 2019-2023 QuestDB
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ ******************************************************************************/
+
+package io.questdb.test.griffin;
+
+import io.questdb.cairo.PartitionBy;
+import io.questdb.griffin.SqlException;
+import io.questdb.std.str.Utf8String;
+import io.questdb.test.AbstractCairoTest;
+import org.junit.Test;
+
+public class AllTablePartitionsTest extends AbstractCairoTest {
+
```

Comment:
I think It would be good to add test with a table being altered or dropped between getCursor() call and actual output assertion, e.g. 

```java
    @Test
    public void testGetAllPartitionsOfMultipleTablesWhenOneGetsDeleted() throws Exception {
        assertMemoryLeak(() -> {
            createTable("a", PartitionBy.DAY, 1);
            createTable("c", PartitionBy.DAY, 2);
            drainWalQueue();

            try (RecordCursorFactory factory = select("select * from all_table_partitions()")) {
                RecordCursor cursor = factory.getCursor(sqlExecutionContext);

                ddl("rename table c to d");
                ddl("rename table d to e");
                drainWalQueue();
                assertCursor("index\tpartitionBy\tname\tminTimestamp\tmaxTimestamp\tnumRows\tdiskSize\tdiskSizeHuman\treadOnly\tactive\tattached\tdetached\tattachable\ttableName\n" +
                        "0\tDAY\t2023-01-02\t2023-01-02T00:00:00.000000Z\t2023-01-02T00:00:00.000000Z\t1\t2097152\t2.0 MiB\tfalse\ttrue\ttrue\tfalse\tfalse\ta\n" +
                        "0\tDAY\t2023-01-02\t2023-01-02T00:00:00.000000Z\t2023-01-02T00:00:00.000000Z\t1\t8\t8.0 B\tfalse\tfalse\ttrue\tfalse\tfalse\tc\n" +
                        "1\tDAY\t2023-01-03\t2023-01-03T00:00:00.000000Z\t2023-01-03T00:00:00.000000Z\t1\t2097152\t2.0 MiB\tfalse\ttrue\ttrue\tfalse\tfalse\tc\n", cursor, factory.getMetadata(), true);
            }

            assertSql("index\tpartitionBy\tname\tminTimestamp\tmaxTimestamp\tnumRows\tdiskSize\tdiskSizeHuman\treadOnly\tactive\tattached\tdetached\tattachable\ttableName\n" +
                    "0\tDAY\t2023-01-02\t2023-01-02T00:00:00.000000Z\t2023-01-02T00:00:00.000000Z\t1\t2097152\t2.0 MiB\tfalse\ttrue\ttrue\tfalse\tfalse\ta\n" +
                    "0\tDAY\t2023-01-02\t2023-01-02T00:00:00.000000Z\t2023-01-02T00:00:00.000000Z\t1\t8\t8.0 B\tfalse\tfalse\ttrue\tfalse\tfalse\te\n" +
                    "1\tDAY\t2023-01-03\t2023-01-03T00:00:00.000000Z\t2023-01-03T00:00:00.000000Z\t1\t2097152\t2.0 MiB\tfalse\ttrue\ttrue\tfalse\tfalse\te\n", "select * from all_table_partitions()");
        });
    }
```